crates/energy/benches/nearby.rs

```diff
@@ -0,0 +1,152 @@
+    World,
+};
+use bevy::time::TimePlugin;
+use criterion::{criterion_group, criterion_main, Criterion};
+use de_core::projection::ToAltitude;
+use de_energy::{update_nearby_recv, EnergyReceiver, NearbyUnits};
+use de_index::{EntityIndex, LocalCollider};
+use de_objects::ObjectCollider;
+use de_test_utils::load_points;
+use parry3d::math::{Isometry, Vector};
+use parry3d::shape::{Cuboid, TriMesh};
+
+const MAP_SIZE: f32 = 2000.;
+const DISTANCE_FROM_MAP_EDGE: f32 = 100.;
+const SPEED: f32 = 10.; // based on MAX_H_SPEED in movement
+
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct UpdateUnits;
+
+#[derive(Component)]
+struct UnitNumber(u32);
+
+#[derive(Component)]
+struct Centre {
+    x: f32,
+    y: f32,
+}
+
+#[derive(Resource)]
+struct Clock(f32); // this clock is used in a substitute of time to make it more deterministic
+
+impl Clock {
+    fn inc(&mut self) {
+        self.0 += 0.008 // 125 updates a "second"
+    }
+}
+
+fn update_index(
+    mut index: ResMut<EntityIndex>,
+    moved: Query<(Entity, &Transform), Changed<Transform>>,
+) {
+    for (entity, transform) in moved.iter() {
+        let position = Isometry::new(
+            transform.translation.into(),
+            transform.rotation.to_scaled_axis().into(),
+        );
+        index.update(entity, position);
+    }
+}
+
+fn init_world_with_entities_moving(world: &mut World, num_entities: u32) {
+    let mut index = EntityIndex::new();
+    let points = load_points(
+        num_entities.try_into().unwrap(),
+        MAP_SIZE - DISTANCE_FROM_MAP_EDGE,
+    );
+
+    for (i, point) in points.into_iter().enumerate() {
+        let point_msl = point.to_msl();
+
+        let collider = LocalCollider::new(
+            ObjectCollider::from(TriMesh::from(Cuboid::new(Vector::new(3., 3., 4.)))),
+            Isometry::new(point_msl.into(), Vector::identity()),
+        );
+
+        let entity = world
+            .spawn((
+                Transform::from_translation(point_msl),
+                Centre {
+                    x: point_msl.x,
+                    y: point_msl.y,
+                },
+                EnergyReceiver,
+                NearbyUnits::default(),
+                UnitNumber(i as u32),
+            ))
+            .id();
+
+        index.insert(entity, collider);
+    }
+
+    world.insert_resource(Clock(0.));
+    world.insert_resource(index);
+}
+
+/// Move entities in circles of radius DISTANCE_FROM_MAP_EDGE / 2.
+fn move_entities_in_circle(
+    clock: Res<Clock>,
+    mut query: Query<(&mut Transform, &UnitNumber, &Centre)>,
+) {
+    for (mut transform, unit_number, centre) in query.iter_mut() {
+        // Change direction (counter)clockwise based on entity_mum % 2 == 0
+        let direction = if unit_number.0 as f32 % 2. == 0. {
+            1.
+        } else {
+            -1.
+        };
+
+        let t = clock.0;
+        let radius = DISTANCE_FROM_MAP_EDGE / 2.;
```

Comment:
We do not have to divide this by two since we keep the center and more at most `radius` from it. Also, I would rename it to `MOVEMENT_RADIUS` which semantically fits more the purpose of the variable (we keep the distance from map edge this large due to the radius of the circular movement and not vice versa).

Note that for a small radius and a large map, accompanying for the radius via the margin has a minimal practical effect. But its more "correct".